kvm/kvm_test.go

```diff
@@ -1379,3 +1379,65 @@ func TestGetSetSRegs2(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestGetSetMSRS(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skipf("Skipping test since we are not root")
+	}
+
+	t.Parallel()
+
+	devKVM, err := os.OpenFile("/dev/kvm", os.O_RDWR, 0o644)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer devKVM.Close()
+
+	vmFd, err := kvm.CreateVM(devKVM.Fd())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	vcpuFd, err := kvm.CreateVCPU(vmFd, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ret, err := kvm.CheckExtension(devKVM.Fd(), kvm.CapGETMSRFeatures)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if int(ret) <= 0 {
+		t.Skipf("Skipping test since CapGETMSRFeatures is disable")
+	}
+
+	// Supported msr indices can be obtained using KVM_GET_MSR_INDEX_LIST in a system ioctl.
+	list := kvm.MSRList{}
+
+	// The first time we probe the number of MSRs.
+	if err := kvm.GetMSRIndexList(devKVM.Fd(), &list); !errors.Is(err, syscall.E2BIG) {
+		t.Fatal(err)
+	}
+
+	// The second time we get the contents of the entries.
+	if err := kvm.GetMSRIndexList(devKVM.Fd(), &list); err != nil {
+		t.Fatal(err)
+	}
+
+	msrs := &kvm.MSRS{
+		NMSRs: list.NMSRs,
+		Entry: kvm.MSREntry{
+			Index: list.Indicies[0],
+		},
```

Comment:
Here, only one entry is given, but list.NMSRs can be greater than 1. Maybe that makes it inconsistent? (I haven't checked that yet)

Reply:
see my comment above.

Reply:
Oh, sorry I mistakenly comment this comment wrongly.

Yes, the number in list.NMSRs can be greater that 1 technically, but the code I looked at does not care I think if there is 1, 100, or anything else, it can only handle one MSREntry I think. It's hard to test to be honest.